dtfs-central-api/src/v1/controllers/utilisation-report-service/put-utilisation-report-status.controller/index.test.ts

```diff
@@ -0,0 +1,302 @@
+    });
+  });
+
+  describe('when trying to only mark reports as not completed', () => {
+    beforeEach(() => {
+      jest.resetAllMocks();
+      jest.mocked(mockTransactionManager.getRepository).mockReturnValue(mockRepository);
+    });
+
+    it('tries to update the correct reports', async () => {
+      // Arrange
+      const reportsWithStatusForMarkingAsNotCompleted: ReportWithStatus[] = [
+        {
+          reportId: 1,
+          status: 'PENDING_RECONCILIATION',
+        },
+        {
+          reportId: 2,
+          status: 'REPORT_NOT_RECEIVED',
+        },
+        {
+          reportId: 3,
+          status: 'REPORT_NOT_RECEIVED',
+        },
+      ];
+
+      const { req, res } = getHttpMocks();
+      req.body.reportsWithStatus = reportsWithStatusForMarkingAsNotCompleted;
+
+      const existingReports = [
+        UtilisationReportEntityMockBuilder.forStatus('RECONCILIATION_COMPLETED')
+          .withId(reportsWithStatusForMarkingAsNotCompleted[0].reportId)
+          .withoutAzureFileInfo()
+          .build(),
+        UtilisationReportEntityMockBuilder.forStatus('RECONCILIATION_COMPLETED')
+          .withId(reportsWithStatusForMarkingAsNotCompleted[1].reportId)
+          .withoutAzureFileInfo()
+          .build(),
+        UtilisationReportEntityMockBuilder.forStatus('RECONCILIATION_COMPLETED').withId(reportsWithStatusForMarkingAsNotCompleted[2].reportId).build(),
+      ];
+
+      const reportIdToStateMachineMap = getReportIdToStateMachineMapFromReports(existingReports);
+
+      jest.spyOn(UtilisationReportStateMachine, 'forReportId').mockImplementation((reportId) => Promise.resolve(reportIdToStateMachineMap[reportId]));
+
+      const createQueryRunnerSpy = jest.spyOn(SqlDbDataSource, 'createQueryRunner').mockReturnValue(mockQueryRunner);
+
+      // Act
+      await putUtilisationReportStatus(req, res);
+
+      // Assert
+      expect(res._getStatusCode()).toBe(HttpStatusCode.Ok);
+      expect(createQueryRunnerSpy).toHaveBeenCalledTimes(1);
+      expect(mockConnect).toHaveBeenCalledTimes(1);
+      expect(mockStartTransaction).toHaveBeenCalledTimes(1);
+      expect(mockCommitTransaction).toHaveBeenCalledTimes(1);
+      expect(mockRollbackTransaction).not.toHaveBeenCalled();
+      expect(mockRelease).toHaveBeenCalled();
+      expect(mockTransactionManager.getRepository).toHaveBeenCalledWith(UtilisationReportEntity);
+      expect(mockRepository.save).toHaveBeenCalledTimes(reportsWithStatusForMarkingAsNotCompleted.length);
+
+      existingReports.forEach((report) => {
+        report.setAsNotCompleted({ requestSource });
+        expect(mockRepository.save).toHaveBeenCalledWith(report);
+      });
+    });
+
+    const reportStatusesToTest: UtilisationReportReconciliationStatus[] = ['PENDING_RECONCILIATION', 'REPORT_NOT_RECEIVED'];
+
+    it.each(reportStatusesToTest)("responds with an error if trying to mark a report with status '%s' as not completed", async (reportStatus) => {
```

Comment:
These statuses are very tied to this test case so could we just inline them in the each?